app/services/idv_attempter.rb

```diff
@@ -1,27 +1,30 @@
 class IdvAttempter
-  attr_accessor :current_user, :user_session
-
-  def initialize(current_user, user_session)
-    self.current_user = current_user
-    self.user_session = user_session
+  def initialize(current_user)
+    @current_user = current_user
   end
 
   def exceeded?
-    # too many attempts in this session
-    return true if idv_attempts >= idv_max_attempts
+    return false if window_expired?
+
+    # too many attempts in the window
+    idv_attempts >= idv_max_attempts && !window_expired?
+  end
 
+  def window_expired?
     # never attempted
     attempted_at = current_user.idv_attempted_at
-    return false unless attempted_at.present?
+    return true  unless attempted_at.present?
 
-    # recent attempts in any session
-    attempted_at + idv_attempt_window > Time.zone.now
+    # last attempted in the past outside the window
+    attempted_at + idv_attempt_window < Time.zone.now
   end
 
   private
 
+  attr_reader :current_user
+
   def idv_attempts
-    user_session[:idv_attempts] || 0
+    current_user.idv_attempts || 0
```

Comment:
Setting a default value of 0 at the DB level would eliminate the need for the `|| 0`, right?